### ISSUE TITLE
Parse housenumber/staircase/unit in Austria

### ISSUE THREAD

**Issue by hfs** [openvenues/libpostal#633]
Hi!

I was checking out libpostal, and saw something that could be improved.

---
#### My country is Austria

---
#### Here's how I'm using libpostal

As part of Pelias, using the [pelias/libpostal-service](https://github.com/pelias/libpostal-service) on Docker.

---
#### Here's what I did

(via the Python package pypostal)

```python
from postal.parser import parse_address
parse_address('Buchengasse 90/3/13, 1100, Wien')
```

---
#### Here's what I got

```python
[('buchengasse', 'road'),
 ('90/3/13', 'house_number'),
 ('1100', 'postcode'),
 ('wien', 'city')]
```
---
#### Here's what I was expecting

The `90/3/13` should be split into house number 90, staircase 3, apartment number 13.

In Austria and especially in Vienna unit numbers are part of the official, administrative addresses and e.g. registered in the land title registry. The numbers are commonly abbreviated as housenumer/unit or housenumber/staircase/unit.

https://de.wikipedia.org/wiki/Wohnungsnummer contains more background information – only in German unfortunately.

Expected output:

```python
[('buchengasse', 'road'),
 ('90', 'house_number'),
 ('3', 'staircase'),
 ('13', 'unit'),
 ('1100', 'postcode'),
 ('wien', 'city')]
```

---
#### For parsing issues, please answer "yes" or "no" to all that apply.

- Does the input address exist in [OpenStreetMap](https://openstreetmap.org)?

No and yes. “Buchengasse 90” is present in https://www.openstreetmap.org/node/1792080332, but the single units are not mapped in OSM.

- Do all the toponyms exist in OSM (city, state, region names, etc.)?

Yes.

- If the address uses a rare/uncommon format, does changing the order of the fields yield the correct result?

No, same result for `90/3/13 Buchengasse, 1100, Wien`

- If the address contains apartment/floor/sub-building information or uncommon formatting, does removing that help? Is there any minimum form of the address that gets the right parse?

```python
parse_address('Buchengasse 90, 1100, Wien')
```

yields the correct

```python
[('buchengasse', 'road'),
 ('90', 'house_number'),
 ('1100', 'postcode'),
 ('wien', 'city')]
```

This allows the address to be found in Pelias (which only knows about the house numbers but not units). But with this workaround I have to check for this pattern in the query string myself. I had hoped that libpostal could do this part for me.

---
#### Here's what I think could be improved

Add training data in Austria/Vienna using this format?

**Comment by albarrentine** [comment]
We do train on that format in Austria (there's some information in the second blog post about staircases, etc.) It will parse them out in long form e.g. "Stiege 3 Wohnung 13" and there may be some other variations/abbreviations included, but the "90/3/13" format can mean too many different things to label one way consistently in the training set. We mostly generate the sub-building information because, as you say, it's often not included in OpenStreetMap, etc. but does exist in written addresses, so the best that can be done in those cases is to generate the missing pieces randomly and label them in ways that will not conflict with hand-labeled strings. For these types of numbering schemes using a "/", it's not only in Austria but in many countries in Latin America, Eastern Europe, Vietnam, etc. and it may mean different things in various countries and may not even be consistent then, so rather than have the model memorize all the local conventions, we opt to label the entire "90/3/13" as a house_number in the training set, and then defer to a downstream task and local knowledge to parse it out further. It can be seen as a form of lossy compression.

For something like geocoding with Pelias, it's easy enough to preprocess the house number token with `h.split("/")[0]` for instance and since the reference record in the search index will likely just be "Buchengasse 90" without the sub-building information anyway, that will usually suffice. Or if the sub-building is needed, again just `h.split("/")` or a regex split will do the trick to recover the remaining components.

**Comment by hfs** [comment]
Thank you, I understand.
